framework/include/loops/ComputeUserObjectsThread.h

```diff
@@ -85,26 +85,29 @@ groupUserObjects(TheWarehouse & w,
                  const std::set<std::string> & aux_deps)
 {
   // Notes about how this information is used later during the simulation:
-  // We only need to run pre-ic objects for their "initial" exec flag time (not the others).
+  // pre-ic objects are run before initial conditions during the "initial" exec flag time.
+  // for any other exec flag time, they are run with the along with the post-aux group.
   //
-  // For pre/post aux objects:
-  //
-  //     If an object was not run as a pre-ic object or it is a pre-ic object and
-  //     shouldDuplicateInitialExecution returns true:
-  //         * run the object at all its exec flag times.
-  //     Else
-  //         * run the object at all its exec flag times *except* "initial"
+  // pre-aux and post-aux objects are run at all exec flag times before and after
+  // auxiliary kernels, respectively.
```

Comment:
I think we could use a bit more explanation here since this is all so tricky.  Maybe some wording similar to the prose on the postaux attribute explanation comment - that has a great explanation, but here is also a spot that benefits from some of that info/context as well.

Reply:
Okay no problem! I will get to this by tonight hopefully...

Reply:
Also, @rwcarlsen since we're talking about `groupUserObjects` here... do you think this part should be an `if` and `else if` statement?

```
    if (ic_deps.count(obj->name()) > 0 ||
        (obj->isParamValid("force_preic") && obj->template getParam<bool>("force_preic")))
    {
      w.update(obj, AttribPreIC(w, true));
      w.update(obj, AttribPostAux(w, false));
    }

    if ((obj->isParamValid("force_preaux") && obj->template getParam<bool>("force_preaux")) ||
        aux_deps.count(obj->name()) > 0 || ic_deps.count(obj->name()) > 0)
    {
      w.update(obj, AttribPreAux(w, true));
      w.update(obj, AttribPostAux(w, false)); 
```

Because, it is technically possible to assign a UO to both the `PRE_IC` and `PRE_AUX` groups by this way. This is how it was before, so I left it this way. It might have made sense before, but now I don't think it does. Unless, for some reason, someone wanted to run their userobject before initial conditions and immediately after.